node/src/accountant/payment_adjuster/non_unit_tests/mod.rs

```diff
@@ -0,0 +1,1258 @@
+    )
+}
+
+fn generate_scenarios(
+    gn: &mut ThreadRng,
+    now: SystemTime,
+    number_of_scenarios: usize,
+) -> Vec<PreparedAdjustmentAndThresholds> {
+    (0..number_of_scenarios)
+        .flat_map(|_| try_making_single_valid_scenario(gn, now))
+        .collect()
+}
+
+fn try_making_single_valid_scenario(
+    gn: &mut ThreadRng,
+    now: SystemTime,
+) -> Option<PreparedAdjustmentAndThresholds> {
+    let accounts_count = generate_non_zero_usize(gn, 25) + 1;
+    let thresholds_to_be_used = choose_thresholds(gn, accounts_count);
+    let (cw_service_fee_balance, qualified_payables, wallet_and_thresholds_pairs) =
+        try_generating_qualified_payables_and_cw_balance(
+            gn,
+            &thresholds_to_be_used,
+            accounts_count,
+            now,
+        )?;
+    let used_thresholds =
+        thresholds_to_be_used.fix_individual_thresholds_if_needed(wallet_and_thresholds_pairs);
+    let analyzed_accounts: Vec<AnalyzedPayableAccount> = convert_collection(qualified_payables);
+    let agent = make_agent(cw_service_fee_balance);
+    let adjustment = make_adjustment(gn, analyzed_accounts.len());
+    let prepared_adjustment = PreparedAdjustment::new(
+        Box::new(agent),
+        None,
+        AdjustmentAnalysis::new(adjustment, analyzed_accounts),
+    );
+    Some(PreparedAdjustmentAndThresholds {
+        prepared_adjustment,
+        used_thresholds,
+    })
+}
+
+fn make_payable_account(
+    idx: usize,
+    thresholds: &PaymentThresholds,
+    now: SystemTime,
+    gn: &mut ThreadRng,
+) -> PayableAccount {
+    // Why is this construction so complicated? Well, I wanted to get the test showing partially
+    // fulfilling adjustments where the final accounts can be paid enough but still not all up to
+    // their formerly claimed balance. It turned out it is very difficult to achieve with the use of
+    // randomized ranges, I couldn't really come up with parameters that would promise this condition.
+    // I ended up experimenting and looking for an algorithm that would make the parameters as random
+    // as possible because the generator alone is not much good at it, using gradually, but
+    // individually generated parameters that I put together for better chances of randomness. Many
+    // produced accounts will not make it through into the actual test, filtered out when attempted
+    // to be converted into a proper QualifiedPayableAccount. This isn't optimal, sure, but it allows
+    // to observe some of those partial adjustments, however, with rather a low rate of occurrence
+    // among those all attempts of acceptable scenarios.
+    let wallet = make_wallet(&format!("wallet{}", idx));
+    let mut generate_age_segment = || {
+        generate_non_zero_usize(
+            gn,
+            (thresholds.maturity_threshold_sec + thresholds.threshold_interval_sec) as usize,
+        ) / 2
+    };
+    let debt_age = generate_age_segment() + generate_age_segment();
```

Comment:
You told that generating the age from starting a number fairly different from 1 isn't right and instead you should start from maturity_threshold_sec.

Instead of dividing by `2`, you'd prefer to multiply it with `2/3` or `3/4`.